qip-0008.md

```diff
@@ -0,0 +1,234 @@
+entropy at block[i]:
+
+$$ \epsilon[i] = S_{uncle}[i-1] * \alpha + S_{uncle}[i]*(1 - \alpha) $$
+
+Next compute the exponentially weighted accepted entropy, $S_{exp}$ at block[i]:
+
+$$ S_{exp}[i] = S[i-1] * \alpha + S[i]*(1 - \alpha) $$
+
+> Note, for efficiency `S` values are recorded and computed in their log2 form
+> (aka bits).
+
+In the above computations, $\alpha$ is a smoothing factor (also referred to as
+TREE_EXPANSION_FILTER_ALPHA), which should be selected to filter our short term
+perturbances in network efficiency. $f$ may be used as an _inverse_ indication
+of the network's security efficiency. The protocol should seek to minimize $f$
+in each zone, which it can do by expanding the tree to create more zones.
+Larger $\alpha$ will increase the total magnitude of uncled entropy required to
+increase $f$. This effectively reduces the protocol's eagerness to expand the
+tree.
+
+##### Computing A Security Score
+The following equation can be used to compute the security score, $f$:
+$$ f[i] = \epsilon[i] / S_{exp}[i] $$
+
+A security score can be computed in the same way for any chain, but this
+protocol is only interested in the global score, denoted as $f_{global}$, to
+decide when to expand the tree of chains. The global score is the $f$ score
+computed on the prime chain.
+
+### Activating New Chains
+When $f_{global}$ exceeds a certain threshold for a long enough time, the
+network can agree that it is time to add more chains to increase the
+transactional capacity of the network. The process for deciding when and how to
+add more chains, is described in the sections below.
+
+#### When to Expand
+The network will agree to add more chains the the tree, once the following
+conditions are met:
+1) all active chains have mined consecutive prime blocks confirming
+$f_{global}$ exceeds the `TREE_EXPANSION_THRESHOLD`
+1) `TREE_EXPANSION_TRIGGER_WINDOW` consecutive prime blocks confirm
+$f_{global}$ after condition 1.
+
+Once the expansion criteria have been met, the network will schedule expansion
+to occur in the future. After `TREE_EXPANSION_WAIT_COUNT` blocks have elapsed since
+the block which finalized the decision to expand, blocks from the new chains
+will be accepted by the network.
+
+#### How Many Chains To Add
+The Quai protocol defines a tree of blockchains of order 2. Initially, the tree
+is of degree 1: prime chain, with a single subordinate region chain, which has
+a single subordinate zone chain. As the network decides to add new chains, it
+will use the following algorithm to decide how many chains to add, and in which
+branches of the tree:
+
+```
+Algorithm to decide which chains to add:
+1: if degree(zone) <= degree(region) then
+2:   add 1 chain to each zone
+3: else then
+4:   add 1 region under prime
+5:   add degree(zone) zones under the new region
+6: end if
+```
+
+#### Old Zones Wait To Send ETXs
+Before an existing chain can send transactions to one of the new chains, the
```

Comment:
This limitation should apply to both the original chains as well as ETXs originating from the newly created chains.

Reply:
good point